src/swarm/neo/client/RequestOnConn.d

```diff
@@ -782,6 +782,11 @@ public class RequestOnConn: RequestOnConnBase, IRequestOnConn
         if (!this.connection)
             return false;
 
+        auto client_connection = cast(Connection)this.connection;
```

Comment:
Looking at the way the [DHT Put request](https://github.com/sociomantic-tsunami/dhtproto/blob/neo-v0.x.x/src/dhtproto/client/request/internal/Put.d#L133-L145) (for example) is implemented, this change would cause the wrong notification to be sent to the user, as far as I can see.

`handler` calls [`NodeHashRanges.putToNode`](https://github.com/sociomantic-tsunami/dhtproto/blob/neo-v0.x.x/src/dhtproto/client/internal/NodeHashRanges.d#L192), which calls `useNode` (the one changed in this PR). If the connection is not up, `useNode` now returns without calling the provided delegate. This causes `handler` to set the result status to `NoNode` (as `put_called` was not set), which causes the `no_node` notification.

We'd actually expect `node_disconnected` to happen, in this situation.

Reply:
Not saying that we shouldn't make this fix, but we might want to investigate the knock-on effects of the change a bit more deeply. It looks like it could lead to pretty subtle and unexpected changes in behaviour.